packages/client-twitter/src/search.ts

```diff
@@ -18,88 +18,99 @@ import { buildConversationThread, sendTweet, wait } from "./utils.ts";
 
+{{providers}}
+
+{{characterPostExamples}}
+
+{{chatDirections}}
 {{postDirections}}
 
 {{recentPosts}}
 
-# Task: Respond to the following post in the style and perspective of {{agentName}} (aka @{{twitterUserName}}). Write a {{adjective}} response for {{agentName}} to say directly in response to the post. don't generalize.
+# Task: Generate a post/reply in the voice, style and perspective of {{agentName}} (@{{twitterUserName}}) while using the thread of tweets as additional context:
+Current Post:
 {{currentPost}}
 
-IMPORTANT: Your response CANNOT be longer than 20 words.
-Aim for 1-2 short sentences maximum. Be concise and direct.
+Thread of Tweets You Are Replying To:
+{{formattedConversation}}
+
+{{actions}}
 
-Your response should not contain any questions. Brief, concise statements only. No emojis. Use \\n\\n (double spaces) between statements.
+# Task: Generate a post in the voice, style and perspective of {{agentName}} (@{{twitterUserName}}). You MUST include an action if the current post text includes a prompt that is similar to one of the available actions mentioned here:
+{{actionNames}}
 
+Here is the current post text again. Remember to include an action if the current post text includes a prompt that asks for one of the available actions mentioned above (does not need to be exact):
+{{currentPost}}
 ` + messageCompletionFooter;
 
-export class TwitterSearchClient extends ClientBase {
+export class TwitterSearchClient  {
+    client: ClientBase;
+    runtime: IAgentRuntime;
     private respondedTweets: Set<string> = new Set();
 
-    constructor(runtime: IAgentRuntime) {
-        // Initialize the client and pass an optional callback to be called when the client is ready
-        super({
-            runtime,
-        });
+    constructor(client: ClientBase, runtime: IAgentRuntime) {
+        this.client = client;
+        this.runtime = runtime;
     }
 
-    async onReady() {
+    async start() {
         this.engageWithSearchTermsLoop();
     }
 
     private engageWithSearchTermsLoop() {
         this.engageWithSearchTerms();
+        const timeoutMinutes = Math.floor(Math.random() * (120 - 60 + 1)) + 60;
+        elizaLogger.log(`Next engagement in ${timeoutMinutes} minutes`);
         setTimeout(
             () => this.engageWithSearchTermsLoop(),
-            (Math.floor(Math.random() * (120 - 60 + 1)) + 60) * 60 * 1000
+            timeoutMinutes * 60 * 1000
         );
     }
 
     private async engageWithSearchTerms() {
-        console.log("Engaging with search terms");
+        elizaLogger.log("Engaging with search terms");
         try {
             const searchTerm = [...this.runtime.character.topics][
                 Math.floor(Math.random() * this.runtime.character.topics.length)
             ];
 
-            console.log("Fetching search tweets");
+            elizaLogger.log("Fetching search tweets");
             // TODO: we wait 5 seconds here to avoid getting rate limited on startup, but we should queue
```

Comment:
update comment time if you're going to change. Also an explanation of why 10s is needed would be helpful